tools/make_icons.py

```diff
@@ -1,60 +1,56 @@
 """
-Generates the toolbar icon images from the FontAwesome font.
+Generates the Matplotlib icon, and the toolbar icon images from the FontAwesome
+font.
 
-First download and extract FontAwesome from http://fontawesome.io/.
-Place the FontAwesome.otf file in the tools directory (same directory
-as this script).
-
-Generates SVG, PDF in one size (size they are vectors) and PNG, PPM and GIF in
-24x24 and 48x48.
+Generates SVG, PDF in one size (since they are vectors), and PNG in 24x24 and
+48x48.
 """
 
-import matplotlib
-matplotlib.use('agg')  # noqa
-
-import os
-
-from PIL import Image
-
-import numpy as np
+from argparse import ArgumentParser, ArgumentDefaultsHelpFormatter
+from io import BytesIO
+from pathlib import Path
+import tarfile
+import urllib.request
 
+import matplotlib as mpl
 import matplotlib.pyplot as plt
 from matplotlib.font_manager import FontProperties
-from matplotlib import cm
-from matplotlib import patheffects
-matplotlib.rcdefaults()
+import numpy as np
 
-matplotlib.rcParams['svg.fonttype'] = 'path'
-matplotlib.rcParams['pdf.fonttype'] = 3
-matplotlib.rcParams['pdf.compression'] = 9
 
+plt.rcdefaults()
+plt.rcParams['svg.fonttype'] = 'path'
+plt.rcParams['pdf.fonttype'] = 3
+plt.rcParams['pdf.compression'] = 9
 
-IMAGES_ROOT = os.path.join(
-    os.path.dirname(__file__), '..', 'lib', 'matplotlib', 'mpl-data', 'images')
-FONT_PATH = os.path.join(
-    os.path.dirname(__file__), 'FontAwesome.otf')
 
+def get_fontawesome():
+    cached_path = Path(mpl.get_cachedir(), "FontAwesome.otf")
+    if not cached_path.exists():
+        with urllib.request.urlopen(
+                "https://github.com/FortAwesome/Font-Awesome"
+                "/archive/v4.7.0.tar.gz") as req, \
+             tarfile.open(fileobj=BytesIO(req.read()), mode="r:gz") as tf:
+            cached_path.write_bytes(tf.extractfile(tf.getmember(
+                "Font-Awesome-4.7.0/fonts/FontAwesome.otf")).read())
+    return cached_path
 
-def save_icon(fig, name):
-    fig.savefig(os.path.join(IMAGES_ROOT, name + '.svg'))
-    fig.savefig(os.path.join(IMAGES_ROOT, name + '.pdf'))
 
+def save_icon(fig, dest_dir, name):
+    fig.savefig(dest_dir / (name + '.svg'))
+    fig.savefig(dest_dir / (name + '.pdf'))
     for dpi, suffix in [(24, ''), (48, '_large')]:
-        fig.savefig(os.path.join(IMAGES_ROOT, name + suffix + '.png'), dpi=dpi)
-
-        img = Image.open(os.path.join(IMAGES_ROOT, name + suffix + '.png'))
-        img.save(os.path.join(IMAGES_ROOT, name + suffix + '.ppm'))
+        fig.savefig(dest_dir / (name + suffix + '.png'), dpi=dpi)
 
 
-def make_icon(fontfile, ccode):
-    prop = FontProperties(fname=fontfile, size=68)
+def make_icon(font_path, ccode):
+    prop = FontProperties(fname=font_path, size=68)
 
     fig = plt.figure(figsize=(1, 1))
     fig.patch.set_alpha(0.0)
     text = fig.text(0.5, 0.48, chr(ccode), ha='center', va='center',
                     fontproperties=prop)
-    text.set_path_effects([patheffects.Normal()])
```

Comment:
I assume this did not have any effect, so removing is fine?

Reply:
The intent was probably to force the text to be rendered as a bezier rather than relying on a font file that may not exist in the end user's computer, but we set svg.fonttype to path so that also does the conversion, and the pdf backend takes care of embedding the font, so we should be good?

Reply:
I don't understand the details here, so relying on your investigation. But your question mark makes me wonder how sure you are...

Reply:
Looking at the code of patheffects.py I'm fairly confident that the description is correct, but to know what the original intent was, we would need to get @mdboom's input, if he still lurks aroud...